### PR TITLE
Cleanup and usage of python API of rosnode

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Maybe best to review commits separately.

### CODE REVIEW

**Review by LarsJanssenTUe — COMMENTED** [review]
LGTM, however I have little to no experience with this piece of code. On the other hand there seem to be no crucial changes. If @PetervDooren has 5 min than maybe he can take a quick look at this?

File: scripts/node_alive_server [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+from std_msgs.msg import String
+import node_alive.srv
+import fnmatch
+from enum import Enum
+from rosnode import rosnode_ping_all
+
+
+class NodeState(Enum):
+    """
+    Enum for representing the node state with a descriptive name
```
Comment (PetervDooren) [inline scripts/node_alive_server:15]: Update docstring as well
```suggestion
    Int for representing the node state with a descriptive name
```

File: package.xml [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
 
-  <run_depend>message_runtime</run_depend>
-  <run_depend>std_msgs</run_depend>
-  <run_depend>rospy</run_depend>
-  <run_depend>diagnostic_msgs</run_depend>
+  <depend>diagnostic_msgs</depend>
+  <depend>std_msgs</depend>
+
+  <exec_depend>message_runtime</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 2">python-enum34</exec_depend>
```
Comment (PetervDooren) [inline package.xml:23]: question: why this dependency on python 2. Would the node-alive software not work with python 3 and up? Or am I misunderstanding this line of code

**Review by PetervDooren — CHANGES_REQUESTED** [review]

File: package.xml (hunk 2 above)
Comment (MatthijsBurgh) [reply package.xml:23]: This is a feature from python 3.4. Currently Ubuntu 18.04 is using 3.6, so that feature is included.

**Review by PetervDooren — APPROVED** [review]

**Pull request merged**